cli-plugins/manager/manager.go

```diff
@@ -48,10 +62,18 @@ func getPluginDirs(dockerCli command.Cli) ([]string, error) {
 
 	pluginDirs = append(pluginDirs, pluginDir)
 	pluginDirs = append(pluginDirs, defaultSystemPluginDirs...)
+	if dockerCli.ClientInfo().HasExperimental {
+		dirs := make([]string, len(pluginDirs), len(pluginDirs)*2)
+		copy(dirs, pluginDirs)
+		for _, dir := range dirs {
+			pluginDirs = append(pluginDirs, dir+"-experimental")
```

Comment:
This puts all experimental paths after all non-experimental paths, rather than interleving them, is that as intended?

IOW this is:
```
~/.docker/cli-plugins
/usr/local/lib/docker/cli-plugins
/usr/lib/docker/cli-plugins
~/.docker/cli-plugins-experimental
/usr/local/lib/docker/cli-plugins-experimental
/usr/lib/docker/cli-plugins-experimental
```

and not

```
~/.docker/cli-plugins
~/.docker/cli-plugins-experimental
/usr/local/lib/docker/cli-plugins
/usr/local/lib/docker/cli-plugins-experimental
/usr/lib/docker/cli-plugins
/usr/lib/docker/cli-plugins-experimental
```

I think it's deliberate but just wanted to check that the implications had been considered i.e. As a user I cannot now install an experimental update to something shipped in `/usr/lib/docker/cli-plugins` in my home dir.

Reply:
Was thinking about this; if someone installs both an  "experimental" and "stable" version of a plugin; which one should take precedence?

Reply:
it should be precisely the first enabled plugin found in the search path, which is why the above distinction on the ordering might matter.

With the current layout I think there is no way to install an experimental version "over" an non-experimental one, other than by pretending it is non-experimental

Reply:
We could do either, I thought this was not only easier to implement, but also made more sense. But happy to change it.